modules/unsupported/cql-json/src/test/java/org/geotools/filter/text/cqljson/CQLJsonParsingTest.java

```diff
@@ -0,0 +1,598 @@
+package org.geotools.filter.text.cqljson;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.geotools.filter.FilterFactoryImpl;
+import org.geotools.filter.text.cql2.CQLException;
+import org.junit.Assert;
+import org.junit.Test;
+import org.opengis.filter.Filter;
+
+public class CQLJsonParsingTest {
+    @Test
+    public void convertJsonToPredicates() throws JsonProcessingException, CQLException {
+        String swimmingpool =
+                "{"
+                        + "             \"and\": ["
+                        + "                {"
+                        + "                   \"eq\": {"
+                        + "                      \"property\": \"swimming_pool\","
+                        + "                      \"value\": true"
+                        + "                   }"
+                        + "                },"
+                        + "                {"
+                        + "                   \"or\": ["
+                        + "                      {"
+                        + "                         \"gt\": {"
+                        + "                            \"property\": \"floor\","
+                        + "                            \"value\": 5"
+                        + "                         }"
+                        + "                      },"
+                        + "                      {"
+                        + "                         \"like\": {"
+                        + "                            \"property\": \"material\","
+                        + "                            \"value\": \"brick%\""
+                        + "                         }"
+                        + "                      },"
+                        + "                      {"
+                        + "                         \"like\": {"
+                        + "                            \"property\": \"material\","
+                        + "                            \"value\": \"%brick\""
+                        + "                         }"
+                        + "                      } "
+                        + "                   ]"
+                        + "                }"
+                        + "             ]"
+                        + "          }";
+
+        CQLJsonCompiler cqlJsonCompiler =
+                new CQLJsonCompiler(swimmingpool, new FilterFactoryImpl());
+        cqlJsonCompiler.compileFilter();
+        Filter filter = cqlJsonCompiler.getFilter();
+        Assert.assertEquals(
+                "[[ swimming_pool = true ] AND [[ floor > 5 ] OR [ material is like brick% ] OR [ material is like %brick ]]]",
+                filter.toString());
+    }
+
+    @Test
+    public void convertAndIntersects() throws JsonProcessingException, CQLException {
+        String beam =
```

Comment:
These methods can be hard to maintain with having to quote content.

* Please see MapBoxTestUtils test for one idea to make this easier, a utility method that changes `\`` to `\"` and then parses.

```
    public static JSONObject object(String json) throws ParseException {
        JSONParser parser = new JSONParser();
        String text = json.replace('\'', '\"');
        Object object = parser.parse(text);
        return (JSONObject) object;
    }
```

Used like:
```
        String jsonStr = "{'name1': 'fred', 'name2' : 1, 'name3' : null, 'name4': { 'age': 1 }}";
        JSONObject object = MapboxTestUtils.object(jsonStr);
```

* Another good approach is to have a property file of reference ECQL expressions to round trip (convert ECQL -parse-> Filter -encode-> CQL-JSON -parse-> Filter and then double check the Filters are the same. Which is what a lot of the duplication in this class is about.

```
convertAndIntersects=(beamMode = "ScanSAR Narrow" ] AND [ swathDirection = "ascending") ...
```

This feedback assume the module will be long lived and plan for adding more and more tests over time as problems are reported. See example [here](https://github.com/geotools/geotools/blob/master/modules/extension/ysld/src/test/java/org/geotools/ysld/parse/YsldParseCookbookTest.java).

Reply:
As suggested I moved the JSON into its own file.